nuxeo-features/nuxeo-automation/nuxeo-automation-server/src/main/java/org/nuxeo/ecm/automation/server/jaxrs/adapters/AsyncOperationAdapter.java

```diff
@@ -0,0 +1,352 @@
+import org.nuxeo.ecm.automation.OperationCallback;
+import org.nuxeo.ecm.automation.OperationContext;
+import org.nuxeo.ecm.automation.OperationException;
+import org.nuxeo.ecm.automation.OperationType;
+import org.nuxeo.ecm.automation.core.impl.InvokableMethod;
+import org.nuxeo.ecm.automation.core.util.BlobList;
+import org.nuxeo.ecm.automation.jaxrs.io.operations.ExecutionRequest;
+import org.nuxeo.ecm.automation.server.AutomationServer;
+import org.nuxeo.ecm.automation.server.jaxrs.OperationResource;
+import org.nuxeo.ecm.automation.server.jaxrs.ResponseHelper;
+import org.nuxeo.ecm.automation.server.jaxrs.RestOperationException;
+import org.nuxeo.ecm.core.api.AsyncService;
+import org.nuxeo.ecm.core.api.AsyncStatus;
+import org.nuxeo.ecm.core.api.Blob;
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.core.api.DocumentModelList;
+import org.nuxeo.ecm.core.api.NuxeoException;
+import org.nuxeo.ecm.core.api.impl.DocumentModelListImpl;
+import org.nuxeo.ecm.core.transientstore.api.TransientStore;
+import org.nuxeo.ecm.core.transientstore.api.TransientStoreService;
+import org.nuxeo.ecm.platform.web.common.exceptionhandling.ExceptionHelper;
+import org.nuxeo.ecm.platform.web.common.vh.VirtualHostHelper;
+import org.nuxeo.ecm.webengine.model.WebAdapter;
+import org.nuxeo.ecm.webengine.model.exceptions.WebResourceNotFoundException;
+import org.nuxeo.ecm.webengine.model.impl.DefaultAdapter;
+import org.nuxeo.runtime.api.Framework;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+/**
+ * @since 10.3
+ */
+@WebAdapter(name = AsyncOperationAdapter.NAME, type = "AsyncOperationAdapter", targetType = "operation")
+@Produces({ MediaType.APPLICATION_JSON })
+public class AsyncOperationAdapter extends DefaultAdapter {
+
+    public static final String NAME = "async";
+
+    private static final String STATUS_STORE_NAME = "automation";
+
+    private String TRANSIENT_STORE_SERVICE = "service";
+
+    private String TRANSIENT_STORE_COMMAND_ID = "commandId";
+
+    private String TRANSIENT_STORE_ERROR = "error";
+
+    private String TRANSIENT_STORE_OUTPUT = "output";
+
+    private String TRANSIENT_STORE_OUTPUT_BLOB = "blob";
+
+    @Context
+    protected HttpServletRequest request;
+
+    @POST
+    public Object doPost(ExecutionRequest xreq) {
+        OperationResource op = (OperationResource) getTarget();
+
+        try {
+            AutomationServer srv = Framework.getService(AutomationServer.class);
+            if (!srv.accept(op.getId(), op.isChain(), request)) {
+                return ResponseHelper.notFound();
+            }
+            final String executionId = UUID.randomUUID().toString();
+
+            // XXX: AsyncContext asyncContext = ctx.getRequest().startAsync();
+            new Thread(() -> {
```

Comment:
Isn't too risky to run a `Thread` without an executor ? I think about a DOS attack on `@async` adapter for instance.